packages/@aws-cdk/cloudformation-diff/lib/diff/index.ts

```diff
@@ -39,35 +45,67 @@ export function diffResource(oldValue?: types.Resource, newValue?: types.Resourc
     const impl = loadResourceModel(resourceType.oldType);
     propertyDiffs = diffKeyedEntities(oldValue!.Properties,
       newValue!.Properties,
-      (oldVal, newVal, key) => _diffProperty(oldVal, newVal, key, impl));
+      (oldVal, newVal, key) => _diffProperty(oldVal, newVal, key, impl, replacement));
 
     otherDiffs = diffKeyedEntities(oldValue, newValue, _diffOther);
     delete otherDiffs.Properties;
+    if (replacement) {
+      delete otherDiffs.Metadata;
+    }
   }
 
   return new types.ResourceDifference(oldValue, newValue, {
     resourceType, propertyDiffs, otherDiffs,
   });
 
-  function _diffProperty(oldV: any, newV: any, key: string, resourceSpec?: Resource) {
-    let changeImpact = types.ResourceImpact.NO_CHANGE;
+  function _diffProperty(
+    oldV: any,
+    newV: any,
+    key: string,
+    resourceSpec?: Resource,
+    changeSetReplacement?: PotentialResourceReplacement,
+  ) {
+    let changeImpact: types.ResourceImpact = types.ResourceImpact.NO_CHANGE;
+    if (!changeSetReplacement) {
+      changeImpact = _resourceSpecImpact(oldV, newV, key, resourceSpec);
+      return new types.PropertyDifference(oldV, newV, { changeImpact });
+    }
+
+    if (changeSetReplacement === RESOURCE_NOT_IN_CHANGE_SET || !(key in changeSetReplacement.propertiesReplaced)) {
+      // The changeset does not contain this property, which means it is not going to be updated. Hide this cosmetic template-only change from the diff
+      return new types.PropertyDifference(1, 1, { changeImpact });
```

Comment:
The 1,1 deliberately obscures the difference between them, since `deepEqual(1,1)` is true.

Reply:
Surely it will not be printed since `changeImpact` is a value that means "don't render this" ?

If your goal was: *"I will put incorrect values into the data model because I know exactly what downstream code will do with these values"*... then congratulations: you've now made a system where different pieces cannot be updated individually anymore, because you need to know how every little piece interacts with every other piece.

I'd rather have pieces that are *locally correct*, and when composed are also *globally correct*.

- It is not *locally correct* to say that the old value and the new value are both `1`. That's lying, and if we try to consume the diff in some other way later on, we'll need to take into account that the model can be lying to us, and we'll need to program in exceptions checking for a value of `1`. (Will we go and refactor later? Fat chance. Nobody dares touching upstream code, especially if it's doing something crazy. Who knows, there might be a reason to do the madness. The poor sap writing a new consumer for this data will dutifully put in a `if (diff.oldV == 1 && diff.newV == 1) { /* do something */ }` and get on with their life, and the madness proliferates).
- If you want to imply certain behavior for downstream systems, and you don't have the fields necessary to express what you want right now, then *add a field*. If changing the interpretation of `changeImpact` doesn't do it for you, it's always better to add `dontRenderThis?: boolean` to the structure than to start putting sentinel values in.

Reply:
thanks for the breakdown, this makes sense. I've gone with the field approach (turns out we already had one, it just wasn't writable)